compiler/prelude/jsmapping.go

```diff
@@ -313,6 +313,13 @@ var $internalize = function(v, t, recv) {
     }
     return s;
   case $kindStruct:
+    if (v !== null && v !== undefined && v.constructor === Date) {
```

Comment:
You can do `!v`, but that matches a few more cases. I'd go with the solution you have.